### PR TITLE
Warnings, fix MSVC warnings

### PR DESCRIPTION
**Pull request by Nekotekina**

### CODE REVIEW

File: rpcs3/Emu/Cell/Modules/sceNpTrophy.cpp [hunk 1]
```diff
@@ -861,7 +861,7 @@ error_code sceNpTrophyUnlockTrophy(u32 context, u32 handle, s32 trophyId, vm::pt
 	const std::string& config_path = vfs::get("/dev_hdd0/home/" + Emu.GetUsr() + "/trophy/" + ctxt->trp_name + "/TROPCONF.SFM");
 	const u32 unlocked_platinum_id = ctxt->tropusr->GetUnlockedPlatinumID(trophyId, config_path);
 
-	if (unlocked_platinum_id != 0u + SCE_NP_TROPHY_INVALID_TROPHY_ID)
+	if (unlocked_platinum_id != umax)
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/sceNpTrophy.cpp]: 😡 at least add a comment with SCE_NP_TROPHY_INVALID_TROPHY_ID at both places

Comment (Nekotekina) [reply rpcs3/Emu/Cell/Modules/sceNpTrophy.cpp]: Okay, I made const unsigned since it's only used here

**Comment by Nekotekina** [comment]
Added formatting for in_addr

**Comment by Nekotekina** [comment]
Generalized strcpy_trunc to support other types, such as string_view.
Added strncpy_trunc to replace std::strncpy.

**Comment by elad335** [comment]
Use the new strcpy_trunc in cellGameGetParamString maybe.

**Comment by KnockPlayz** [comment]
This update broke Ratchet and Clank: Going Commando USA with HDD error. Downgrading fixed the issue.

**Comment by RainbowCookie32** [comment]
#7694 probably fixed it

**Pull request merged**